### ISSUE TITLE
Fix S4015 FP: Non inherited method with generic parameters is falsely detected as inherited

### ISSUE THREAD

**Issue by FelixDamrau** [SonarSource/sonar-dotnet#8666]
### Description

Rule: S4015 
If you derive from a class that has a generic class as an attribute in a virtual method, you cannot use a method with the same name and different generic attributes.

### Repro steps

You can use this code to reproduce the issue
```cs
public class DerivedClass : BaseClass
{
    private void DoSomething(IEnumerable<string> numbers) { /* ... */ }
}

public class BaseClass
{
    protected virtual void DoSomething(IEnumerable<int> numbers) { }
}
```

### Expected behavior

S4015 is not raised for `private void DoSomething(IEnumerable<string> numbers)`  as it does not hide anything of the base class.

### Actual behavior

S4015 is raised.
The different method-signature is even shown in the popup-dialog:
![vs-popup](https://github.com/SonarSource/sonar-dotnet/assets/20964228/3757f920-3a21-4891-9e56-4f2cdef9c156)

### Known workarounds

None of the suggestested workarounds can be used.
Sealing the class supresses the issue (although it should not, as mentioned in #1189 )
Changing the access modifier suppresses the issue (although it should not, as it does not solve the hiding issue)

I cannot seal the class, nor can I change the access modifier.

### Related information

* Visual Studio Plugin: SonarLint for Visual Studio 2022, Version: 7.6.0.83110
* Visual Studio version: 17.8.4
* MSBuild / dotnet version:  17.8.3/8.0.101
* SonarScanner for .NET version (if used): -
* Operating System: Windows 11 Pro, 23H2, 22631.3007

**Comment by pavel-mikula-sonarsource** [comment]
Internal ticket [NET-1621](https://sonarsource.atlassian.net/browse/NET-1621)

[NET-1621]: https://sonarsource.atlassian.net/browse/NET-1621?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
